blockchain/vm/evm.go

```diff
@@ -256,7 +261,7 @@ func (evm *EVM) Call(caller types.ContractRef, addr common.Address, input []byte
 	)
 
 	// Filter out invalid precompiled address calls, and create a precompiled contract object if it is not exist.
-	if IsPrecompiledContractAddress(addr, evm.chainRules) {
+	if IsPrecompiledContractAddress(addr, evm.chainRules) || (addr == consoleLogContractAddress && evm.Config.UseConsoleLog) {
 		precompiles := evm.GetPrecompiledContractMap(caller.Address())
```

Comment:
Modify the GetPrecompiledContractMap instead?

Reply:
```
// Because IsPrecompiledContractAddress checks for 0..0x400 and ConsoleLog address is outside of the range, we add one more condition if UseConsoleLog.
```

Reply:
Added, thanks! 01fd418